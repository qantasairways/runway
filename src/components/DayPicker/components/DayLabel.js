import React from 'react';
import PropTypes from 'prop-types';
import { css } from 'emotion';

import { colours, fontWeight, mq } from '../../../theme/airways';

function DayLabelStyles({ leftAligned, bottomAligned, isSelected }) {
  const verticalOffset = isSelected ? '-3px' : 0;

  return css({
    label: 'runway-calendar__day-label',
    display: 'flex',
    flexDirection: leftAligned ? 'row' : 'row-reverse',
    alignItems: 'center',
    position: 'absolute',
    left: leftAligned ? 0 : 'initial',
    right: leftAligned ? 'initial' : 0,
    top: bottomAligned ? 'initial' : verticalOffset,
    bottom: bottomAligned ? verticalOffset : 'initial',
    height: '14px',
    boxSizing: 'border-box',
    paddingTop: isSelected && !bottomAligned ? '3px' : 0,
    paddingRight: leftAligned ? 0 : '2px',
    paddingLeft: leftAligned ? '2px' : '0',
    fontSize: '0.625rem',
    fontWeight: fontWeight.bold,
    lineHeight: 0,
    background: colours.hightlightsLight,
    '&::after': {
      content: '""',
      position: 'absolute',
      top: 0,
      right: leftAligned ? '-5px' : 'initial',
      left: leftAligned ? 'initial' : '-5px',
      border: 'solid transparent',
      borderRight: leftAligned
        ? 'solid transparent'
        : `solid ${colours.hightlightsLight}`,
      borderLeft: leftAligned
        ? `solid ${colours.hightlightsLight}`
        : 'solid transparent',
      borderWidth: leftAligned ? '7px 0 7px 5px' : '7px 5px 7px 0'
    },
    [mq.medium]: {
      height: '16px',
      fontSize: '0.75rem',
      paddingRight: leftAligned ? 0 : '3px',
      paddingLeft: leftAligned ? '3px' : 0,
      '&::after': {
        right: leftAligned ? '-6px' : 'initial',
        left: leftAligned ? 'initial' : '-6px',
        borderWidth: leftAligned ? '8px 0 8px 6px' : '8px 6px 8px 0'
      }
    }
  });
}

function DayLabel({ leftAligned, bottomAligned, isSelected, label, Icon }) {
  return label ? (
    <div css={DayLabelStyles({ leftAligned, bottomAligned, isSelected })}>
      <span>{label}</span>
      {Icon && (
        <Icon
          className={css({
            marginBottom: '1px',
            fill: colours.darkerGrey,
            transform: leftAligned ? 'none' : 'scaleX(-1)'
          })}
          height="10"
          width="10"
        />
      )}
    </div>
  ) : null;
}

DayLabel.propTypes = {
  leftAligned: PropTypes.bool,
  bottomAligned: PropTypes.bool,
  isSelected: PropTypes.bool,
  label: PropTypes.string,
  Icon: PropTypes.func
};

DayLabel.defaultProps = {
  leftAligned: true,
  bottomAligned: false,
  isSelected: false,
  label: '',
  Icon: null
};

export default DayLabel;
