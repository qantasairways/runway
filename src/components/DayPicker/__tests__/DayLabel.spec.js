import React from 'react';
import { shallow } from 'enzyme';
import DayLabel from '../components/DayLabel';

const DummyIcon = () => <div>Icon</div>;

describe('DayLabel', () => {
  let component;

  it('renders correctly with defaults', () => {
    component = shallow(<DayLabel />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly with props provided', () => {
    component = shallow(
      <DayLabel
        leftAligned={false}
        bottomAligned
        isSelected
        label="Label"
        Icon={DummyIcon}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
