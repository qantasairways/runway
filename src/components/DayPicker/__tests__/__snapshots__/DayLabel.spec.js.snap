// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayLabel renders correctly with defaults 1`] = `""`;

exports[`DayLabel renders correctly with props provided 1`] = `
.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  left: initial;
  right: 0;
  top: initial;
  bottom: -3px;
  height: 14px;
  box-sizing: border-box;
  padding-top: 0;
  padding-right: 2px;
  padding-left: 0px;
  font-size: 0.625rem;
  font-weight: 700;
  line-height: 0;
  background: #bff4f2;
}

.emotion-1::after {
  content: "";
  position: absolute;
  top: 0;
  right: initial;
  left: -5px;
  border: solid transparent;
  border-right: solid #bff4f2;
  border-left: solid transparent;
  border-width: 7px 5px 7px 0;
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    height: 16px;
    font-size: 0.75rem;
    padding-right: 3px;
    padding-left: 0;
  }

  .emotion-1::after {
    right: initial;
    left: -6px;
    border-width: 8px 6px 8px 0;
  }
}

.emotion-0 {
  margin-bottom: 1px;
  fill: #323232;
  -webkit-transform: scaleX(-1);
  -ms-transform: scaleX(-1);
  transform: scaleX(-1);
}

<div
  className="emotion-1"
>
  <span>
    Label
  </span>
  <DummyIcon
    className="emotion-0"
    height="10"
    width="10"
  />
</div>
`;
